static/app/components/smartSearchBar/index.tsx

```diff
@@ -328,11 +329,30 @@ class SmartSearchBar extends React.Component<Props, State> {
     savedSearchType: SavedSearchType.ISSUE,
   };
 
+  static getDerivedStateFromProps(props: Props, state: State) {
+    const {query} = props;
+    const {lastPropQuery} = state;
+
+    if (!defined(query)) {
+      return null;
+    }
+
+    if (query === lastPropQuery) {
```

Comment:
Is `addSpace(query)` very expensive? 

This is pedantic but IMO it defeats the purpose of a static method when it's not idempotent. I think `componentDidUpdate(prevProps, prevState)` might be better.